DKDBManager/NSManagedObject+DKDBManager.h

```diff
@@ -18,32 +38,308 @@ typedef NS_ENUM(NSInteger, DKDBManagedObjectState) {
+ * This function allows you to check that and avoid the update of the current entity during the CRUD process.
+ *
+ * For example you might want to verify the `lastUpdate` attribute or a version number from the given dictionary against the current entity.
+ * If they match return then `false` to NOT update the entity and avoid some useless actions.
+ *
+ * The result of this function is ignored when the debug states of the manager are set as
+ * `needForcedUpdate == true` or if `allowUpdate == false`.
+ *
+ * @return TRUE if the entity should be updated by the manager; default is TRUE.
+ */
+ - (BOOL)shouldUpdateEntityWithDictionary:(NSDictionary *)dictionary;
+
+/**
+ * @brief Override this function to enable all activitiy logs about the current class entity or not.
+ *
+ * @return TRUE if log activated; default is FALSE.
+ */
+ + (BOOL)verbose;
+
+/**
+ * @brief Override this function to set a default sorting key for the current class entity.
+ *
+ * @return Sorting key as a NSString object; default is nil.
+ */
+ + (NSString *)sortingAttributeName;
+
+/**
+ * @brief Override this function to create a predicate that will find the right entity corresponding to the given dictionary.
+ *
+ * @param dictionary A NSDictionary object containing information about the database entity to fetched with.
+ *
+ * @discussion This function is called during the CRUD process to fetch an entity from the database.
+ * Depending on the returned predicate the manager will update an existing entity or will create a new one.
+ *
+ * If returns nil then will only take the first entity created (if any) and update it. By doing so only ONE entity will ever be created.`
+ *
+ * If returns a `false predicate` then a new entity will always be created.
+ *
+ * Otherwise use the entity found by the predicate.
+ * The predicate should be created depending on the parameter: `dictionary`.
+ *
+ * For example, your entity `Book` could be fetch through its `releaseDate` and `title` attributes.
+ *
+ * @return A NSPredicate object to find/fetch an entity in the local database.
+ */
+ + (NSPredicate *)primaryPredicateWithDictionary:(NSDictionary *)dictionary;
+
+/**
+ * @brief Returns all entities of the current class model.
+ *
+ * @discussion The values are sorted by the default sorting attribute.
+ *
+ * @see + (NSString *)sortingAttributeName;
+ *
+ * @return A NSArray object containing all entities from the current class model.
+ */
 + (NSArray *)all;
-+ (NSInteger)count;
+
+/**
+ * @brief Count all entities of the current class model.
+ *
+ * @return A NSInteger value corresponding to the total number of entities of the current class model.
+ */
+ + (NSInteger)count;
 
 #pragma mark - UPDATE
 
-- (void)updateWithDictionary:(NSDictionary *)dict;
+/**
+ * @brief Override to update the current entity with a given dictionary.
+ *
+ * @param dictionary A NSDictionary object containing information about the database entity to fetched with.
```

Comment:
it should say something like: Dictionary with informations to update the entity.